tests/python_package_test/test_engine.py

```diff
@@ -2887,3 +2888,37 @@ def hook(obj):
     dumped_model_str = str(bst.dump_model(5, 0, object_hook=hook))
     assert "leaf_value" not in dumped_model_str
     assert "LV" in dumped_model_str
+
+
+def test_force_split_with_feature_fraction(tmp_path):
+    def check_force_splits(dumped_model):
```

Comment:
Since this is called only one time, inside this one test case, I don't think there's any added benefit to encapsulating this code in a function. To eliminate some indirection and make the test a bit easier for others to understand, can you please just include this code at the bottom of this test case?

```python
# ensure every tree used this split
tree_info = gbm.dump_model()["tree_info"]
assert len(tree_info) > 1
for tree in tree_info:
    tree_structure = tree["tree_structure"]
    assert tree_structure['split_feature'] == 0
```

Reply:
Thanks, updated.

Reply:
Thanks! Can you please also include the `assert len(tree_info) > 1` from my suggestion?

That's just a little extra defense in case `gbm.dump_model()["tree_info"]` is unexpectedly empty. If that was empty, none of the code under `for tree in tree_info` would be executed.

Reply:
Thanks, updated.